lib/cryptoexchange/exchanges/trust_dex/services/market.rb

```diff
@@ -0,0 +1,55 @@
+require 'bigdecimal'
+
+module Cryptoexchange::Exchanges
+  module TrustDex
+    module Services
+      class Market < Cryptoexchange::Services::Market
+        class << self
+          def supports_individual_ticker_query?
+            false
+          end
+        end
+
+        def fetch
+          output = super(ticker_url)
+          adapt_all(output)
+        end
+
+        def ticker_url
+          "#{Cryptoexchange::Exchanges::TrustDex::Market::API_URL}/return_ticket.php"
+        end
+
+        def adapt_all(output)
+          tickers = []
+          output.map do |pair|
+            pair.each do |key, value|
+              target, base = key.split("_")
+              market_pair = Cryptoexchange::Models::MarketPair.new({
+                              base: base,
+                              target: target,
+                              market: TrustDex::Market::NAME
+                            })
+              tickers << adapt(value, market_pair)
+            end
+          end
+
+          tickers
+        end
+
+        def adapt(output, market_pair)
+          ticker           = Cryptoexchange::Models::Ticker.new
+          ticker.base      = market_pair.base
+          ticker.target    = market_pair.target
+          ticker.market    = TrustDex::Market::NAME
+          ticker.last      = NumericHelper.to_d(output['last'])
+          ticker.high      = NumericHelper.to_d(output['highestBid'])
+          ticker.low       = NumericHelper.to_d(output['lowestAsk'])
+          ticker.volume    = NumericHelper.to_d(output['baseVolume'])
```

Comment:
@Swingcloud Since the volume should refer to the base currency, it should be `quoteVolume` right?

Reply:
😂 seems like I forgot to change